app/src/main/java/com/example/ide/data/local/DeviceAppRepository.kt

```
@@ -0,0 +1,80 @@
+                    appName = appInfo.loadLabel(packageManager).toString(),
+                    versionName = pkg.versionName,
+                    versionCode = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+                        pkg.longVersionCode
+                    } else {
+                        @Suppress("DEPRECATION")
+                        pkg.versionCode.toLong()
+                    },
+                    isSystemApp = isSystem,
+                    apkPath = appInfo.sourceDir
+                )
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)